### PR TITLE
add input 'actual_shape' to bilinear_interp_op

### PR DESCRIPTION
**Pull request by heavengate**
add a Variable param in ' actual_shape' image_size python api, if 'actual_shape' is set, use actual_shape data as OutSize to Specify output dims.

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5570,6 +5570,7 @@ def dice_loss(input, label, epsilon=0.00001):
 def image_resize(input,
                  out_shape=None,
                  scale=None,
+                 actual_shape=None,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 1.  兼容性： `actual_shape`要放在最后
2. 需要改resize_bilinear接口。

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -5586,13 +5587,19 @@ def image_resize(input,
         input (Variable): The input tensor of image resize layer,
                           This is a 4-D tensor of the shape
                           (num_batches, channels, in_h, in_w).
-        out_shape(list|tuple|Variable|None): Output shape of image resize
+        out_shape(list|tuple|None): Output shape of image resize
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 考虑到兼容性，out_shape依然要是Variable，但warning，建议用户通过`actual_shape`来设置。

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -5586,13 +5587,19 @@ def image_resize(input,
...
         scale(float|None): The multiplier for the input height or width.
                          At least one of out_shape or scale must be set.
                          And out_shape has a higher priority than scale.
                          Default: None
+        actual_shape(Variable): An optional input. If provided, image resize 
+                                according to this given shape rather than 
+                                :attr:`out_shape` and :attr:`scale` specifying
+                                shape. That is to say actual_shape has the 
+                                highest priority.
+                                Default: None
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py]: 再加上使用场景，比如output shape每个batch动态可变时通过该参数设置，同时告诉用户设置还要设置out_shape，否则gprah构造时可能出错。

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (heavengate) [reply python/paddle/fluid/layers/nn.py]: Done

File: python/paddle/fluid/layers/nn.py (hunk 3 above)
Comment (heavengate) [reply python/paddle/fluid/layers/nn.py]: Done

**Review by heavengate — COMMENTED** [review]
Thahks for reviewing and have done for all suggestions

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -5623,19 +5644,28 @@ def _is_list_or_turple_(data):
...
-            raise ValueError('out_shape should be a list or tuple or variable')
-        if _is_list_or_turple_(out_shape):
-            out_shape = list(map(int, out_shape))
-            out_h = out_shape[0]
-            out_w = out_shape[1]
-        else:
+        if isinstance(out_shape, Variable):
+            print("Warning: out_shape as Variable type is deprecated, \
+                    it is recommended to use actual_shape instead of \
+                    out_shape to specify output shape dynamically.")
```
Comment (qingqing01) [inline python/paddle/fluid/layers/nn.py:5651]: 在上面：

`import warnings`

用：
```
warnings.warn('xxx')
```

Comment (heavengate) [reply python/paddle/fluid/layers/nn.py:5651]: Done.

**Review by qingqing01 — APPROVED** [review]

**Comment by heavengate** [comment]
Since blinear_interp_op was integrated into interpolate_op, these changes have been merged in https://github.com/PaddlePaddle/Paddle/pull/14205

**Pull request closed without merging**